src/routes/docs/tutorials/apple/step-4/+page.markdoc

```diff
@@ -0,0 +1,193 @@
+            .setProject("[YOUR_PROJECT_ID]")
+        self.account = Account(client)
+        self.database = Databases(client)
+    }
+    
+    
+    public func currentSession() async throws -> User<[String: AnyCodable]> {
+        try await account.get()
+    }
+    
+    public func onRegister(
+        _ email: String,
+        _ password: String
+    ) async throws -> User<[String: AnyCodable]> {
+        try await account.create(
+            userId: ID.unique(),
+            email: email,
+            password: password
+        )
+    }
+    
+    public func onLogin(
+        _ email: String,
+        _ password: String
+    ) async throws -> Session {
+        try await account.createEmailSession(
+            email: email,
+            password: password
+        )
+    }
+    
+    public func onLogout() async throws {
+        _ = try await account.deleteSession(
+            sessionId: "current"
+        )
+    }
+    
+    
+    public func onAccountDelete() async throws {
+        _ = try await account.deleteSession(
+            sessionId: "current"
+        )
+    }
+    
+    public func getDocs<T>(_ db: Database, _ collection: DatabaseCollections, queries: [String]? = nil) async throws -> DocumentList<T> {
+        try await database.listDocuments<T>(
+            databaseId: db.rawValue,
+            collectionId: collection.rawValue,
+            queries: queries,
+            nestedType: T.self
+        )
+    }
+    
+    public func insertDoc(_ db: Database, _ collection: DatabaseCollections, data: Any) async throws {
+        _ = try await database.createDocument(
+            databaseId: db.rawValue,
+            collectionId: collection.rawValue,
+            documentId: ID.unique(),
+            data: data
+        )
+    }
+    
+    
+    public func updateDoc(_ db: Database, _ collection: DatabaseCollections, _ id: String, data: Any) async throws {
+        _ = try await database.updateDocument(
+            databaseId: db.rawValue,
+            collectionId: collection.rawValue,
+            documentId: id,
+            data: data
+        )
+    }
+    
+    
+    public func removeDoc(_ db: Database, _ collection: DatabaseCollections, _ id: String, data: Any) async throws {
+        _ = try await database.deleteDocument(
+            databaseId: db.rawValue,
+            collectionId: collection.rawValue,
+            documentId: id
+        )
+    }
+    
+}
+```
+
+# Snackbar service {% #snackbar-service %}
+
+In order to manage error toast, we need a view model to handle and dispatch error messsgae to our UI. Create a new file SnackbarService.swift inside **Shared/Services** folder and add the following code to it.
+
+```swift
+struct SnackBarState: Identifiable, Equatable {
+    static func == (lhs: SnackBarState, rhs: SnackBarState) -> Bool {
+        lhs.id == rhs.id
+    }
+    
+    let id = UUID()
+    let hasError: Bool
+    let error: Error
+}
+
+class SnackBarService: ObservableObject {
```

Comment:
<img width="1015" alt="Screenshot 2023-10-25 at 3 50 53 PM" src="https://github.com/appwrite/website/assets/29069505/49163ef1-5093-45c2-a3af-566216ff44b1">
Could we be missing some dependencies?